pkg/test/echo/server/forwarder/protocol.go

```diff
@@ -71,6 +76,40 @@ func newProtocol(cfg Config) (protocol, error) {
 	timeout := common.GetTimeout(cfg.Request)
 	headers := common.GetHeaders(cfg.Request)
 
+	var getClientCertificate func(info *tls.CertificateRequestInfo) (*tls.Certificate, error)
+	if cfg.Request.Cert != "" && cfg.Request.Key != "" {
+		cert, err := tls.X509KeyPair([]byte(cfg.Request.Cert), []byte(cfg.Request.Key))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse x509 key pair: %v", err)
+		}
+
+		for _, c := range cert.Certificate {
+			cert, err := x509.ParseCertificate(c)
+			if err != nil {
+				log.Errorf("Failed to parse client certificate: %v", err)
+			}
+			log.Infof("Using client certificate [%s] issued by %s", cert.SerialNumber, cert.Issuer)
+			for _, uri := range cert.URIs {
+				log.Infof("  URI SAN: %s", uri)
+			}
+		}
+		getClientCertificate = func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
Do we only set this for adding addition logging info? This seems like it may be pretty spammy at info level?

Reply:
yes, otherwise we could use the `Certificates` field. I find this information useful for debugging especially when using custom root CA. Changed to log.Debugf